pkg/vmprovider/providers/vsphere/session/session_vm_update.go

```diff
@@ -402,20 +402,29 @@ func UpdateHardwareConfigSpec(
 	configSpec *vimTypes.VirtualMachineConfigSpec,
 	vmClassSpec *v1alpha1.VirtualMachineClassSpec) {
 
-	// TODO: Looks like a different default annotation gets set by VC.
-	if config.Annotation != constants.VCVMAnnotation {
-		configSpec.Annotation = constants.VCVMAnnotation
-	}
 	if nCPUs := int32(vmClassSpec.Hardware.Cpus); config.Hardware.NumCPU != nCPUs {
 		configSpec.NumCPUs = nCPUs
 	}
 	if memMB := virtualmachine.MemoryQuantityToMb(vmClassSpec.Hardware.Memory); int64(config.Hardware.MemoryMB) != memMB {
 		configSpec.MemoryMB = memMB
 	}
+}
+
+func UpdateConfigSpecAnnotation(
+	config *vimTypes.VirtualMachineConfigInfo,
+	configSpec *vimTypes.VirtualMachineConfigSpec) {
+	if config.Annotation != constants.VCVMAnnotation {
+		configSpec.Annotation = constants.VCVMAnnotation
+	}
+}
+
+func UpdateConfigSpecManagedBy(
```

Comment:
A few notes:

* Let's switch the order of the params, and
* use `dst` and `src` as they make sense in this instance.
* Let's also introduce `defaultValue string` in case `dst.Annotation` is `nil`. The value of `defaultValue` can be the value of the VM Class's ConfigSpec managedBy value.

```
func UpdateConfigSpecManagedBy(
	dst *vimTypes.VirtualMachineConfigSpec,
	src vimTypes.VirtualMachineConfigInfo,
	defaultValue *vimTypes.ManagedByInfo) {

	// If the VM was created the constant value, then do not
	// ever update the value again.
	if src.ManagedBy != nil && 
		src.ManagedBy.ExtensionKey == constants.ManagedByExtensionKey && 
		src.Type == constants.ManagedByExtensionType {

		return
	}

	// Update the destination's managedBy if it is not already set.
	if dst.ManagedBy == nil {
		if defaultValue != nil {
			dst.ManagedBy = defaultValue
		} else {
			dst.ManagedBy =  &vimTypes.ManagedByInfo{
				ExtensionKey: constants.ManagedByExtensionKey,
				Type:         constants.ManagedByExtensionType,
			}
		}
	}
}
```

Reply:
@aruneshpa and I discussed, we cannot allow VM Class to own ManagedBy due to iconography and telemetry.